src/wiki/templates/wiki/base_site.html

```diff
@@ -39,7 +39,14 @@
           {% endblock %}
         </div>
         <div class="navbar-collapse collapse">
-          <form class="navbar-form navbar-right" id="wiki-search-form" method="GET" action="{% url 'wiki:search' %}">
+          <form class="navbar-form navbar-right" id="wiki-search-form" method="GET" action="{% spaceless %}
+              {% if article or urlpath %}
+                {% url 'wiki:search' article_id=article.id path=urlpath.path %}
+              {% else %}
+                {% url 'wiki:search' %}
+              {% endif %}
+            {% endspaceless %}"
+          >
```

Comment:
Oh interesting! From a user perspective though, this might get confusing because you will be searching in the current hierarchy but without an option to search the full wiki.

This will also be a behavioral change for anyone used to using this for searching the full wiki.

Could we fix this by editing the placeholder text for the input field? For instance `Search in '{{ article.title }}'` vs `Search full wiki` ?

Also, all of this behavior should maybe be conditioned by whether or not there actually exist any nested articles? Because if not, we might as well default to searching globally?

Reply:
I agree. I think we should clearly label both the search input and the search view.

Also what about adding a checkbox to the search view to force full search? Reddit does it like this:
![image](https://user-images.githubusercontent.com/1637715/33015906-c2dbfb30-cdec-11e7-8e9f-ab114e1c5b5d.png)

Reply:
>Also, all of this behavior should maybe be conditioned by whether or not there actually exist any nested articles? Because if not, we might as well default to searching globally?

Right now the search also includes the article you searched from, so I think this would seem odd to the user. It think it makes sense to allow searching for text in just one article, even though it's probably easier to just do a CTRL+F search.

Reply:
> Right now the search also includes the article you searched from

That's cool, I mean at least not catastrophic in any way..

> Also what about adding a checkbox to the search view to force full search? Reddit does it like this:

To me, it's not mandatory to "expand" the task, I think as long as we make sure the user can easily see where she is searching, it's all fine. But it would be nice with some additional features for the search area, just keep in mind that there's limited space in the topbar, and it has to be responsive.. so I don't know? Perhaps a separate PR?